File: go/grains/grains.go
package grains

import (
	"fmt"
)

//Square calculates the number of grains on a given square of a chessboard.
func Square(n int) (uint64, error) {
	if n <= 0 || n > 64 {
		return 0, fmt.Errorf("Out of acceptable range")
	}
	return 1 << uint64(n-1), nil
}

//Total calculates the number of grains on a chessboard.
func Total() uint64 {
	return 1<<64 - 1

}

File: python/grains/grains.py
def square(number):
    if number > 64 or number < 1:
        raise ValueError("Number must be greater than one and less than or equal to 64")
    return 2 ** (number - 1)


def total(number):
    if number > 64 or number < 1:
        raise ValueError("Number must be greater than one and less than or equal to 64")
    return sum([square(i+1) for i in range(number)])
